src/hooks/useTableSort.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useMemo, useState } from "react";
+import { ColumnProps } from "../components/Table/types";
+
+export function useTableSort<T>(data: T[], columns: ColumnProps<T>[]) {
+  const [sortStates, setSortStates] = useState<{
+    [key: string]: { direction: "asc" | "desc" | null; priority: number };
+  }>({});
+
+  const handleColumnSort = useCallback((columnKey: string, sorter?: (a: T, b: T) => number) => {
+    if (!sorter) return;
+
+    setSortStates((prev) => {
+      const currentState = prev[columnKey]?.direction ?? null;
+      const nextDirection: "asc" | "desc" | null =
+        currentState === null ? "asc" : currentState === "asc" ? "desc" : null;
+
+      if (nextDirection === null) {
+        return Object.fromEntries(
+          Object.entries(prev)
+            .filter(([key]) => key !== columnKey)
+            .map(([key, value]) => [
+              key,
+              { ...value, priority: value.priority > prev[columnKey].priority ? value.priority - 1 : value.priority },
+            ]),
+        );
+      }
+
+      const highestPriority = Math.max(0, ...Object.values(prev).map((s) => s.priority));
+      return {
+        ...prev,
+        [columnKey]: { direction: nextDirection, priority: highestPriority + 1 },
+      };
+    });
+  }, []);
+
+  const sortedData = useMemo(() => {
+    const activeSorters = Object.entries(sortStates)
+      .filter(([, state]) => state.direction !== null)
+      .sort((a, b) => b[1].priority - a[1].priority)
```

Comment:
use `toSorted` instead

Reply:
do we update our compiler target? beause our ver doesnt support that

Reply:
Yeah, good catch, check here https://stackoverflow.com/questions/76710595/how-can-i-use-es2023-methods-in-typescript